Userland/Services/WindowServer/Window.cpp

```diff
@@ -916,7 +916,8 @@ bool Window::hit_test(const Gfx::IntPoint& point, bool include_frame) const
     u8 threshold = alpha_hit_threshold() * 255;
     if (threshold == 0 || !m_backing_store || !m_backing_store->has_alpha_channel())
         return true;
-    auto color = m_backing_store->get_pixel(point.translated(-rect().location()));
+    auto relative_point = point.translated(-rect().location()) * Screen::the().scale_factor();
```

Comment:
afaik no windows have scaled backing stores yet. i'd expect this to do the wrong thing. if you want to future-proof, you can get the scale factor off m_backing_store itself though (`m_backing_store->scale()`).

Reply:
Hm good point. I was also wondering why Bitmap::get_pixel didn't use the scale factor it had.

Reply:
Maybe it should! My thinking was that it returns a single pixel, so it kind of has to be pixel-based. Maybe we should have a "average_rect_sample()" or what that gets a sample rect size defaulting to 1x1 and that could then work in logical coords and return the 2x2 avg on 2x bitmaps.